vine/src/features/type_.rs

```diff
@@ -67,14 +67,32 @@ impl<'core> Charter<'core, '_> {
 
 impl<'core> Resolver<'core, '_> {
   pub(crate) fn resolve_type_alias(&mut self, alias_id: TypeAliasId) {
-    if let Some(Some(_)) = self.sigs.type_aliases.get(alias_id) {
-      return;
+    match self.sigs.type_aliases.get(alias_id) {
+      Some(TypeAliasState::Resolved(_)) => return,
+      Some(TypeAliasState::Resolving) => {
+        // Cycle detected
+        let alias_def = &self.chart.type_aliases[alias_id];
+        self.core.report(Diag::RecursiveTypeAlias { span: alias_def.ty.span });
```

Comment:
`report` is called later in `let error_type` so this currently duplicates the error message
```suggestion
```